modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import 'ckeditor4';
+
+export function BalloonEditor({
+	config,
+	contents,
+	name,
+}: IBalloonEditorProps): JSX.Element;
+
+export function ClassicEditor({
+	contents,
+	editorConfig,
+	onChange,
+	ref,
+}: IClassicEditorProps): JSX.Element;
+
+export interface IEditor {
+	editor: CKEDITOR.editor;
+}
+
+interface IBalloonEditorProps {
+	config?: CKEDITOR.config;
```

Comment:
There are many places in this PR where the value of property interface is `CKEDITOR.config` or `CKEDITOR.editor`. This does not look right. Value should be a primitive, a `type` or another `interface`, right?

Reply:
> CKEDITOR.config
In cases we're setting editor config

> CKEDITOR.editor
For cases where we're receving editor?

I'm not understanding you, could you bring more details?

Reply:
This is a TypeScript `interface`. `interface` determines object property _types_, not _values_. `string` is an example of type. `CKEDITOR.config` and `CKEDITOR.editor` are values. It's probably good enough to say `config: object` on this line, as those objects are too complex to drill down into.

I am not an expert on TS, so I'm not 100% sure if this is some shorthand way of setting types, but it _looks_ wrong. By the book, types of objects properties should be [primitives, a `type` or another `interface`](https://www.typescriptlang.org/docs/handbook/2/objects.html).

Reply:
>  not values

but `config` in this case is another interface https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ckeditor4/index.d.ts#L585
![image](https://user-images.githubusercontent.com/7663513/184009988-43cfa0fe-8967-46f5-9450-3e46a5e917e0.png)

and `editor` is a [namespace on definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ckeditor4/index.d.ts#L1081) and can be used to get more types for editor stuff

I'm not understanding you when you say values 😅 

>  It's probably good enough to say config: object on this line, as those objects are too complex to drill down into.

but how these types can be more reliable? I'll remove ckeditor types then and continue with the simpler object to not block this PR

> I am not an expert on TS

me neither, we are together 🤝🫂

Reply:
>config in this case is another interface

Oh, I see. Yeah, this is fine then :+1: